packages/backend/src/controllers/house/house.controller.ts

```diff
@@ -100,22 +112,27 @@ export class HouseController {
     @Body() joinHouseDto: JoinHouseDto,
     @User() user: DecodedIdToken,
   ): Promise<HouseResponseDto> {
+    const addedUser = await this.userStoreService.findOneByFirebaseId(user.uid);
     const house = await this.houseStoreService.findOneByCode(
       joinHouseDto.houseCode,
     );
-
     if (house != null) {
+      house.users.push(addedUser._id);
       await this.userStoreService.updateByFirebaseId(user.uid, {
         house: house._id,
       });
+      await this.houseStoreService.update(house.id, {
+        users: house.users,
+      });
       const owner = await this.userStoreService.findOne(house.owner);
-
+      const userList = await this.houseStoreService.getUserDto(house.id);
```

Comment:
This should be changed according to the above comment

Reply:
If you get the house with population on line 116 then you can just set the users on line 135 to `house.users`

Reply:
Currently the UserResponseDto is not extending UserModel or UserDocument, so I think you can't just set the users to be populated house.users? 

Also the house on line116 is found by houseCode while the ````GET /house```` is thru Id, so unless you are suggesting the following way:
````
return await this.HouseModel.findOne({$or: [{ _id: id }, {code: code}]}).populate('users').exec();
````
with optional parameters, I think we would need to create two functions here.

Reply:
I don't think it will be an issue if you make the UserResponseDto extend the UserModel. I would try it anyway before you do this. The fields are the same so it might not be an issue (I am not a typescript expert). Just play around with it a bit and see what works. 

Also I meant to say create two functions, one for find by code and one for find by id. That is also fine in my opinion because this will improve the code and reduce the complexity by using populate instead of the loops.

Reply:
I just tried it out and the house in UserReponseDto is a string representing the houseCode, and the house in UserModel is an ObjectID reference to the House.

Reply:
Can you make the house in UserResponseDto an ObjectId?

Reply:
I can but that would also require changing multiple previous implementations. @j-chad Any thoughts?

Reply:
I would prefer if the api returns the code. The internal ID is not useful to the front-end